pandas/core/reshape/tile.py

```diff
@@ -230,7 +230,7 @@ def cut(x, bins, right=True, labels=None, retbins=False, precision=3,
         else:
             bins = np.asarray(bins)
         bins = _convert_bin_to_numeric_type(bins, dtype)
-        if (np.diff(bins) < 0).any():
+        if (np.diff(bins.astype('float64')) < 0).any():
```

Comment:
Can you add a comment above this line indicating why this is being done, something like:
```python
bins = _convert_bin_to_numeric_type(bins, dtype)

# GH 26045: cast to float to avoid an overflow
if (np.diff(bins.astype('float64')) < 0).any():